### ISSUE TITLE
Unannotated fields should be ignored for Dataclass

### ISSUE THREAD

**Issue by yellowjello** [microsoft/pyright#710]
**Describe the bug**
According to PEP 557 (https://www.python.org/dev/peps/pep-0557/#id7), only variables with annotations are considered fields for dataclass purposes. However, Pyright considers all variables including those that don't have annotations for the data class field default value check as well as the constructor.

**To Reproduce**
1. Enter the code in the below section into VSCode with Pyright extension.
2. Pyright will say that there is an issue with `Thing2` with the message `Data class fields without default value cannot appear after data fields with default values`.
3. There is also no error in the `Thing2(a=1, b=3, c="dsf")` line.

**Expected behavior**
There should not be an error with `c` since `b` in Thing is not a dataclass field as it does not have an annotation.

In addition, there should be error in the `Thing2(a=1, b=3, c="dsf")` line since `b` should not be part of the constructor for `Thing2` (again because `b` is not a dataclass field).

**Screenshots or Code**
```
from dataclasses import dataclass
@dataclass
class Thing:
    a: int
    b = 5

@dataclass
class Thing2(Thing):
    c: str

Thing2(a=1, b=3, c="dsf")
```

**VS Code extension or command-line**
VS Code extension version 1.1.40

**Comment by erictraut** [comment]
Thanks for the bug report. I missed that fine print in PEP 557. Thankfully, the fix is easy — it involves deleting five lines of code that specifically handled the unannotated case. :)

This will be addressed in the next version of Pyright.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.41, which I just published.
